modules/apps/portal-search/portal-search-rest-impl/src/main/java/com/liferay/portal/search/rest/internal/resource/v1_0/SuggestionResourceImpl.java

```diff
@@ -217,6 +224,41 @@ else if (StringUtil.equals(scope, "this-site") && (groupId != null)) {
 		searchContext.setTimeZone(contextUser.getTimeZone());
 		searchContext.setUserId(contextUser.getUserId());
 
+		long commerceChannelGroupId = 0;
+
+		CommerceChannel commerceChannel =
+			_commerceChannelLocalService.fetchCommerceChannelBySiteGroupId(
+				groupId);
+
+		if (commerceChannel != null) {
+			commerceChannelGroupId = commerceChannel.getGroupId();
+		}
+
+		if (commerceChannelGroupId > 0) {
+			AccountEntry accountEntry =
+				_commerceAccountHelper.getCurrentAccountEntry(
+					commerceChannelGroupId, themeDisplay.getRequest());
+
+			searchContext.setAttribute(
+				"accountEntryId", accountEntry.getAccountEntryId());
+
+			long[] commerceAccountGroupIds = null;
+
+			if (accountEntry != null) {
+				commerceAccountGroupIds =
+					_accountGroupLocalService.getAccountGroupIds(
+						accountEntry.getAccountEntryId());
+			}
+
+			searchContext.setAttribute(
+				"commerceAccountGroupIds", commerceAccountGroupIds);
+
+			searchContext.setAttribute(
+				CPField.COMMERCE_CHANNEL_GROUP_ID, commerceChannelGroupId);
```

Comment:
I think this should go first as it's a constant (also before the accountEntry definition)